### PR TITLE
test: 7 c tests from others

### PR DESCRIPTION
**Pull request by mintbomb27**
## Related Issue(s)
#546 

## Description
Contains 7 more tests covering all tests with names starting with c in others.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [ ] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

**Review by chipzoller — CHANGES_REQUESTED** [review]

File: other/check-subjectaccessreview/02-setup-user.yaml
[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/check-subjectaccessreview/02-setup-user.yaml:1]: Sure! Will do that. And yes, will make it more generic.

File: other/check-subjectaccessreview/policy-ready.yaml
[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/check-subjectaccessreview/policy-ready.yaml:1]: Sorry! Will delete that.

File: other/check-serviceaccount/01-assert.yaml
[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/check-serviceaccount/01-assert.yaml:1]: I changed the match clause to this and the `badpod01` passed.
```yaml

      match:
        any:
        - resources:
            kinds:
            - Pod
          subjects:
          - kind: ServiceAccount
            name: "?*"
            namespace: "*"
```

File: other/check-env-vars/pods-bad.yaml
[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/check-env-vars/pods-bad.yaml:1]: Done!

File: other/check-serviceaccount/01-assert.yaml
[the remark this answers is not in the document]
Comment (chipzoller) [reply other/check-serviceaccount/01-assert.yaml:1]: What if the `namespace` field is omitted?

[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/check-serviceaccount/01-assert.yaml:1]: The `name` as well as the `namespace` field are both required.

[the remark this answers is not in the document]
Comment (chipzoller) [reply other/check-serviceaccount/01-assert.yaml:1]: Maybe try `"?*"` in there?

[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/check-serviceaccount/01-assert.yaml:1]: No change, same result.

[the remark this answers is not in the document]
Comment (chipzoller) [reply other/check-serviceaccount/01-assert.yaml:1]: Can you post in the kyverno channel on Kubernetes Slack with a reproduction? Would like to see why this isn't working.

[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/check-serviceaccount/01-assert.yaml:1]: I'm not sure how this should be working? Are wildcards allowed in `subjects`?

[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/check-serviceaccount/01-assert.yaml:1]: https://kubernetes.slack.com/archives/CLGR9BJU9/p1687809575945679

[the remark this answers is not in the document]
Comment (mintbomb27) [reply other/check-serviceaccount/01-assert.yaml:1]: What should I do about this? Just let it be?

[the remark this answers is not in the document]
Comment (chipzoller) [reply other/check-serviceaccount/01-assert.yaml:1]: Yes, just leave this as-is.

**Review by chipzoller — APPROVED** [review]

**Pull request merged**